regolith/helpers/makeappointmentshelper.py

```diff
@@ -0,0 +1,134 @@
+    #subpi.add_argument("-o", "--out_of_date", action="store_true",
+    #                   help='get group members supported on an out of date grant')
+    #subpi.add_argument("-d", "--depleted", action="store_true",
+    #                   help='get group members supported on a depleted grant')
+    subpi.add_argument("-b", "--begin_date", help='begin date of interval to check appointment, to specify')
+    subpi.add_argument("-e", "--end_date", help='end date of interval to check appointments')
+    subpi.add_argument("-o", "--overspent", action="store_true", help='return grants that are overspent by more' 
+                                                                      'than 10k by their end')
+    subpi.add_argument("-u", "--underspent", action="store_true", help='return grants that are underspent by more' 
+                                                                       'than 10k by their end')
+    subpi.add_argument("-m", "-member", help='suggest appointments for this member')
+    subpi.add_argument("-a", "--appoint", action="store_true", help='suggest new appointments')
+
+    return subpi
+
+
+class MakeAppointmentsHelper(SoutHelperBase):
+    """Helper for managing appointments.
+
+       Appointments are...
+    """
+    # btype must be the same as helper target in helper.py
+    btype = HELPER_TARGET
+    needed_dbs = [f'{TARGET_COLL}', "grants"]
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        if "groups" in self.needed_dbs:
+            rc.pi_id = get_pi_id(rc)
+        rc.coll = f"{TARGET_COLL}"
+        try:
+            if not rc.database:
+                rc.database = rc.databases[0]["name"]
+        except:
+            pass
+        colls = [
+            sorted(
+                all_docs_from_collection(rc.client, collname), key=_id_key
+            )
+            for collname in self.needed_dbs
+        ]
+        for db, coll in zip(self.needed_dbs, colls):
+            gtx[db] = coll
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+
+    def sout(self):
+        rc = self.rc
+        gaps = []
+        outdated = []
+        depleted = []
+        if rc.underspent:
+            underspent =[]
+        if rc.overspent:
+            overspent = []
+        if rc.begin_date:
+            begin_date = date_parser.parse(rc.begin_date).date()
+        else:
+            begin_date = dt.date.today()
+        if rc.end_date:
+            end_date = date_parser.parse(rc.end_date).date()
+        else:
+            end_date = begin_date + relativedelta(years=1)
+
+        for person in self.gtx["people"]:
+            appts = person.get("appointments")
+            if not is_fully_appointed(appts, begin_date=rc.begin_date, end_date=rc.end_date)[0]:
```

Comment:
shouldn't this be ``begin_date=begin_date`` instead of ``begin_date=rc.begin_date`` (and so on)

Reply:
done